compiler/rustc_hir_analysis/src/hir_ty_lowering/lint.rs

```diff
@@ -83,7 +86,15 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
             // Check if the impl trait that we are considering is an impl of a local trait.
             self.maybe_suggest_blanket_trait_impl(self_ty, &mut diag);
             self.maybe_suggest_assoc_ty_bound(self_ty, &mut diag);
-            diag.stash(self_ty.span, StashKey::TraitMissingMethod);
+            if let Some(mut sugg) =
```

Comment:
please leave a comment here explaining why we need to do this. afaik it's for the case where there's something that could be both an associated type or a trait object

Reply:
Done. Workflow needs approval again